### PR TITLE
Programmatically remove EnginePtr/inject, EXPAND_Engine, and WITH_ENGINE_TEST_HELPER*

### PR DESCRIPTION
**Pull request by racer-coder**
for i in ": public EnginePtr " ", public EnginePtr" "EXPAND_Engine;" "EXPAND_Engine"; do
    git grep -l "$i" | xargs sed -i "s/$i//g"
done

git grep -l "inject" | xargs sed -i "/inject[(][)]/d"

for i in WITH_ENGINE_TEST_HELPER_SENS WITH_ENGINE_TEST_HELPER_BOARD_CALLBACK WITH_ENGINE_TEST_HELPER; do
    git grep -l "$i" | xargs sed -i "s/$i/EngineTestHelper eth/g"
done

git checkout firmware/controllers/core/engine_ptr.h
git checkout unit_tests/global.h

### CODE REVIEW

File: firmware/controllers/trigger/trigger_central.cpp [hunk 1]
```diff
@@ -52,10 +52,8 @@ TriggerCentral::TriggerCentral() : trigger_central_s(),
 }
 
 void TriggerCentral::init() {
```
Comment (mck1117) [inline firmware/controllers/trigger/trigger_central.cpp:54]: empty function

File: firmware/controllers/algo/engine.cpp [hunk 2]
```diff
@@ -463,13 +462,8 @@ void Engine::OnTriggerSyncronization(bool wasSynchronized) {
 #endif
 
 void Engine::injectEngineReferences() {
-	triggerCentral.inject();
-	primaryTriggerConfiguration.inject();
 	for (int camIndex = 0;camIndex < CAMS_PER_BANK;camIndex++) {
```
Comment (mck1117) [inline firmware/controllers/algo/engine.cpp:465]: empty loop

**Comment by racer-coder** [comment]
This was fine until I removed some dead code :-P

**Comment by mck1117** [comment]
That HW CI failure is a known fluke, you're ok!

**Pull request merged**